stakingModel/contracts/UBIScheme.sol

```diff
@@ -264,31 +300,27 @@ contract UBIScheme is AbstractUBI {
         requireActive
         returns (bool)
     {
-        require(accounts.length < gasleft().div(iterationGasLimit), "exceeds of gas limitations");
+        require(
+            accounts.length < gasleft().div(iterationGasLimit),
+            "exceeds of gas limitations"
+        );
         for(uint256 i = 0; i < accounts.length; ++i) {
-            fish(accounts[i]);
+            if(isRegistered(accounts[i]) && !isActiveUser(accounts[i]) && !fishedUsersAddresses[accounts[i]]) {
+                fish(accounts[i]);
+            }
         }
         return true;
     }
 
-    /* @dev Function for automate claiming UBI for users. Emits the caller address and the
-     * given list length and the actual number of claimers.
-     * @param accounts - claimers account list
-     * @return A bool indicating if UBI was claimed
-     */
-    function distribute(address[] memory accounts)
+    /**
+    * @dev making the contract inactive after it has transferred funds to `_avatar`
+    * only the avatar can destroy the contract.
+    * @param _avatar destination avatar address for funds
+    */
+    function end(Avatar _avatar)
```

Comment:
empty end function makes no sense, also i've said that there's nothing to implement just add a test

Reply:
updated
3342c9c0a5e4ba58dae711a4431d0da78fc5094e